api/api.proto

```diff
@@ -3763,6 +3782,42 @@ message SdkVolumeSnapshotScheduleUpdateRequest {
 message SdkVolumeSnapshotScheduleUpdateResponse {
 }
 
+// Defines the request to watch an openstorage event. An event can be a volume, a node, a disk, etc
+message SdkWatchRequest {
+  // event_type is an option to indicate the type of openStorage event to watch on.
+  // examples of options are volume, node, disk, etc
+  oneof event_type {
+    // watch request for volume event
+    SdkVolumeWatchRequest volume_event = 1;
+  }
+}
+
+// Defines the response to watch an openstorage event. An event can be a volume, a node, a disk, etc
+message SdkWatchResponse {
+  // event_type is an option to indicate the type of openStorage event to watch on.
+  // examples of options are volume, node, disk, etc
+  oneof event_type {
+    SdkVolumeWatchResponse volume_event = 1;
+  }
+}
+
+// Defines the request to watch an openstorage volume event for the given label
+// if the label is empty, it returns all the volume events
+message SdkVolumeWatchRequest {
+  // Options during watch
+  map<string, string> labels = 1;
+}
+
+// Defines the response containing an volume with a state changed
+message SdkVolumeWatchResponse {
+  // Information about the volume
+  Volume volume = 1;
+  // Name of volume
+  string name = 2;
+  // Volume labels
+  map<string, string> labels = 3;
```

Comment:
Do we need to return the labels separately? The labels are already a part of Volume

Reply:
there could be additional labels that were not part of the request